File: scribbleshare-room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/server/messages/ServerMessageCanvasUpdate.java
package net.stzups.scribbleshare.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.data.objects.canvas.canvasUpdate.CanvasUpdate;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageCanvasUpdate extends ServerMessage {
    private final CanvasUpdate[] canvasUpdates;

    public ServerMessageCanvasUpdate(CanvasUpdate[] canvasUpdates) {
        this.canvasUpdates = canvasUpdates;
    }

    @Override
    protected ServerMessageType getMessageType() {
        return ServerMessageType.CANVAS_UPDATE;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeByte((byte) canvasUpdates.length);
        for (CanvasUpdate canvasUpdate : canvasUpdates) {
            canvasUpdate.serialize(byteBuf);
        }
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageCanvasUpdate.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import ByteBuffer from "../../ByteBuffer.js";
import CanvasUpdates from "../../../canvas/canvasUpdate/CanvasUpdates.js";

export default class ServerMessageCanvasUpdate extends ServerMessage {
    canvasUpdates: Array<CanvasUpdates>;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.canvasUpdates = [];
        let length = byteBuffer.readUint8();
        for (let i = 0; i < length; i++) {
            this.canvasUpdates.push(new CanvasUpdates(byteBuffer));
        }
    }

    getType(): ServerMessageType {
        return ServerMessageType.CANVAS_UPDATE;
    }
}